src/app-layer-detect-proto.c

```diff
@@ -2041,6 +2093,15 @@ AppProto AppLayerProtoDetectGetProtoByName(const char *alproto_name)
 {
     SCEnter();
 
+    AppLayerProtoDetectAliases *cur_alias = alpda_ctx;
+    while (cur_alias != NULL) {
```

Comment:
Why did you use a linked list ?

Reply:
My intention was to add a generic functionality that parsers can register arbitrary aliases, the linked list pattern is used often in suricata and i don't know before how many aliases will be registered....

Reply:
An array would have seemed simpler to me, but that's fine I guess